feature/settings/src/main/java/com/google/samples/apps/nowinandroid/feature/settings/SettingsViewModel.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.nowinandroid.feature.settings
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.google.samples.apps.nowinandroid.core.data.repository.UserDataRepository
+import com.google.samples.apps.nowinandroid.core.model.data.DarkThemeConfig
+import com.google.samples.apps.nowinandroid.core.model.data.ThemeBrand
+import com.google.samples.apps.nowinandroid.feature.settings.SettingsUiState.Loading
+import com.google.samples.apps.nowinandroid.feature.settings.SettingsUiState.Success
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class SettingsViewModel @Inject constructor(
+    private val userDataRepository: UserDataRepository,
+) : ViewModel() {
+    val settingsUiState: StateFlow<SettingsUiState> =
+        userDataRepository.userDataStream
+            .map { userData ->
+                Success(
+                    settings = UserEditableSettings(
+                        brand = userData.themeBrand,
+                        darkThemeConfig = userData.darkThemeConfig
+                    )
+                )
+            }
+            .stateIn(
+                scope = viewModelScope,
+                // Starting eagerly means the user data is ready when the SettingsDialog is laid out
+                // for the first time. Without this the layout is done using the "Loading" text,
+                // then replaced with the user editable fields once loaded, however, the layout
+                // height doesn't change meaning all the fields are squashed into a small
+                // scrollable column.
```

Comment:
I think we're running into https://issuetracker.google.com/issues/221643630 here.

Reply:
Good to know, thanks. I tried the workaround suggested in the bug which was to add 

```
properties = DialogProperties(usePlatformDefaultWidth = false)
```

But this resulted in the dialog filling all the available width and looked pretty strange. I've added a note to remove the current workaround when the bug is fixed.